inference-engine/src/mkldnn_plugin/nodes/mkldnn_convert_node.cpp

```diff
@@ -0,0 +1,92 @@
+        outDims.push_back(MKLDNNDims(output.getDims()));
+    if (inDims.empty() && input.getLayout() != InferenceEngine::Layout::ANY)
+        inDims.push_back(MKLDNNDims(input.getDims()));
+    if (getParentEdges().size() != 1)
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << "Incorrect number of output edges for layer " << getName();
+}
+
+void MKLDNNConvertNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto layer = getCnnLayer();
+    if (layer == nullptr) {
+        THROW_IE_EXCEPTION << "Cannot get CNN layer for layer name " << getName();
+    }
+
+    LayerConfig config;
+    DataConfig dataIn;
+    DataConfig dataConfigOut;
+
+    memory::format fmt;
+
+    if (input.getLayout() != InferenceEngine::Layout::ANY && output.getLayout() != InferenceEngine::Layout::ANY) {
+        dataIn.desc = input;
+        config.inConfs.push_back(dataIn);
+
+        const auto layout = config.inConfs[0].desc.getLayout(); // inp/out layouts must be the same
+        dataConfigOut.desc = TensorDesc(output.getPrecision(), input.getDims(), layout);
+        config.outConfs.push_back(dataConfigOut);
+        fmt = MKLDNNMemory::Convert(layout);
+    } else if (layer->insData.size() == 1 && layer->outData.size() == 1) {
+        const SizeVector& ins_dims = layer->insData[0].lock()->getTensorDesc().getDims();
+        const auto layout = layer->insData[0].lock()->getTensorDesc().getLayout(); // inp/out layouts must be the same
+        dataIn.desc = TensorDesc(layer->insData[0].lock()->getTensorDesc().getPrecision(), ins_dims, layout);
+        config.inConfs.push_back(dataIn);
+
+        const SizeVector& out_dims = layer->outData[0]->getTensorDesc().getDims();
+        dataConfigOut.desc = TensorDesc(layer->outData[0]->getTensorDesc().getPrecision(), out_dims, layout);
+        config.outConfs.push_back(dataConfigOut);
+        fmt = MKLDNNMemory::Convert(layout);
+    } else {
+        THROW_IE_EXCEPTION << "Convert layer with name '" << getName() << "' has incorrect number of input/output edges";
+    }
+
+    config.dynBatchSupport = false;
+    supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown, fmt);
+}
+
+void MKLDNNConvertNode::createPrimitive() {
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        THROW_IE_EXCEPTION << "Destination memory didn't allocate.";
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+        THROW_IE_EXCEPTION << "Input memory didn't allocate.";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        THROW_IE_EXCEPTION << "Preferable primitive descriptor is not set.";
+}
+
+void MKLDNNConvertNode::execute(mkldnn::stream strm) {
+    auto& parentMem = getParentEdgeAt(0)->getMemory();
```

Comment:
We have to add starting offset (offset to the first element in the block).

Reply:
Done in e1639f394ed3299b1def7fd04e3484aaeda0f5cd.